pkg/cmd/server/start/start_master.go

```diff
@@ -699,20 +699,66 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 
 	glog.Infof("Started Kubernetes Controllers")
 
-	// no special order
+	openshiftControllerContext := origincontrollers.ControllerContext{
+		KubeControllerContext: controllerContext,
+		ClientBuilder: origincontrollers.OpenshiftControllerClientBuilder{
+			ControllerClientBuilder: controller.SAControllerClientBuilder{
+				ClientConfig:         restclient.AnonymousClientConfig(&oc.PrivilegedLoopbackClientConfig),
+				CoreClient:           oc.PrivilegedLoopbackKubernetesClientsetExternal.Core(),
+				AuthenticationClient: oc.PrivilegedLoopbackKubernetesClientsetExternal.Authentication(),
+				Namespace:            bootstrappolicy.DefaultOpenShiftInfraNamespace,
+			},
+		},
+		DeprecatedOpenshiftInformers: oc.Informers,
+		Stop: controllerContext.Stop,
+	}
+	openshiftControllerInitializers, err := oc.NewOpenshiftControllerInitializers()
+
+	allowedOpenshiftControllers := sets.NewString(
+		"deployer",
+		"deploymentconfig",
+		"deploymenttrigger",
+	)
 	if configapi.IsBuildEnabled(&oc.Options) {
-		err := oc.RunBuildController(oc.Informers)
+		allowedOpenshiftControllers.Insert("build")
+	}
+
+	if err != nil {
+		glog.Errorf("Could not start build controller: %v", err)
+		return err
+	}
+
+	for controllerName, initFn := range openshiftControllerInitializers {
+		// TODO remove this.  Only call one to start to prove the principle
+		if !allowedOpenshiftControllers.Has(controllerName) {
+			glog.Warningf("%q is skipped", controllerName)
+			continue
+		}
+		if !openshiftControllerContext.IsControllerEnabled(controllerName) {
+			glog.Warningf("%q is disabled", controllerName)
+			continue
+		}
+
+		glog.V(1).Infof("Starting %q", controllerName)
+		started, err := initFn(openshiftControllerContext)
 		if err != nil {
-			glog.Fatalf("Could not start build controller: %v", err)
+			glog.Errorf("Error starting %q", controllerName)
```

Comment:
Huh, I thought I made this process death.  Should fatal here I think to allow us to crashloop.